### PR TITLE
Add birthday field and validation

### LINKED ISSUE #115: V1.4 MVP: Birthday feature
Add birthday for each contact.

Tag for birthday as `b/` 

Add date validation.

### LINKED ISSUE #124: V1.4 Feature: Add birthday into View Window
Birthday information has not been added into the view window.

### PR DESCRIPTION
**Pull request by zjaoyuki**
- Closes #115

**Changes:**
- Added b/ prefix for birthday field in contact entries
- Implemented date validation with dd-MM-yyyy format (e.g., 24-12-2005)
- Updated all relevant components: Model, Storage, UI, and Tests
- Birthday field is displayed in person cards and view window

**Key Features:**
- Mandatory field when adding new contacts
- Proper date validation prevents invalid entries (will add more restrictions in v1.5)
- Full test coverage including edge cases
- Backward compatible with existing data

**EDIT:**
- Closes #124 (integrated birthday field into view window)

The birthday feature is now fully integrated and ready for use.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Birthday.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Person's birthday in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBirthday(String)}
+ */
+public class Birthday {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Birthday should be in the format dd-MM-yyyy (e.g., 24-12-2005) and must be a valid date";
```
Comment (TheresiaOng) [inline src/main/java/seedu/address/model/person/Birthday.java]: I think we previously agreed on the "yyyy-MM-dd" format? Perhaps we could apply the same here?

File: src/test/java/seedu/address/model/person/BirthdayTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    @Test
+    public void isValidBirthday() {
+        // null birthday
+        assertThrows(NullPointerException.class, () -> Birthday.isValidBirthday(null));
+
+        // invalid birthdays
+        assertFalse(Birthday.isValidBirthday("")); // empty string
+        assertFalse(Birthday.isValidBirthday(" ")); // spaces only
+        assertFalse(Birthday.isValidBirthday("32-13-2005")); // invalid date
+        // assertFalse(Birthday.isValidBirthday("31-04-2020")); // April only has 30 days
```
Comment (TheresiaOng) [inline src/test/java/seedu/address/model/person/BirthdayTest.java]: Great test cases! But are these meant to be commented out?

Comment (zjaoyuki) [reply src/test/java/seedu/address/model/person/BirthdayTest.java]: Yeps, I'll implement it in the next iteration. It's stricter date time validation

File: src/test/java/seedu/address/model/person/BirthdayTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        assertFalse(Birthday.isValidBirthday("15-03-abcd")); // non-numeric year
+        assertFalse(Birthday.isValidBirthday("00-03-2020")); // zero day
+        assertFalse(Birthday.isValidBirthday("15-00-2020")); // zero month
+        assertFalse(Birthday.isValidBirthday("15-13-2020")); // month > 12
+        assertFalse(Birthday.isValidBirthday("32-01-2020")); // day > 31
+
+        // valid birthdays
+        assertTrue(Birthday.isValidBirthday("15-03-2018")); // valid date
+        assertTrue(Birthday.isValidBirthday("24-12-2017")); // valid date
+        assertTrue(Birthday.isValidBirthday("29-02-2020")); // leap year
```
Comment (deepa-m1) [inline src/test/java/seedu/address/model/person/BirthdayTest.java]: Interesting test cases considered such as leap year!

**Review by deepa-m1 — APPROVED** [review]
Good job on integrating with UI files! Perhaps could consider making test cases more succinct.

**Review by TheresiaOng — APPROVED** [review]
Good job on implementing the birthday feature!
